### ISSUE TITLE
SSRS 2017 - Report Search failure - HTTP 401 (unauthorized)

### ISSUE THREAD

**Issue by erindempster** [microsoft/Reporting-Services#144]
I've implemented the 2017 custom security extension, and 95% of the functionality is working properly.  The one item that is not working is the search tool in the web portal.  When any user of the system tries to search for reports, a HTTP 401 (unauthorized) error is logged, but with no supporting information.

Has anyone else seen this behavior and what was done to fix it?  Again, it doesn't matter if the user is a Report Browser, Publisher or System Admin.  The HTTP 401 error is logged when the search functionality is used.  In the browser, it is announced an error has occurred, momentarily, before being dropped back to the Forms Authentication login page.

**Comment by erindempster** [comment]
Has anyone encountered something like this in the past?

Any ideas on how to resolve this?

**Comment by jtarquino** [comment]
Did you check in the logs in the portal for example "C:\Program Files\Microsoft Power BI Report Server\PBIRS\LogFiles\RSPortal_2019_04_29_14_51_14.log"

I just tried with the sample with the default configuration with the script and it works properly

**Comment by erindempster** [comment]
I did look at the RSPortalxxx.log file.  I can see the successful loading of the top-level folder (I even pulled the URL out of the log to confirm that's what the command is returning.)  I also see the receive event and its 401 status.  I don't see anything else in this log to point to an issue.  

In trying to find my search parameter, I even started up SQL Profiler and captured the data around the search.  My search term was not found in the Profiler data, which I would have expected to see in a call to search the thousands of items for reports named with my search term.

I've compared Authorization.cs from the GitHub repo with the Authorization.cs file in my project and they are identical, with the exception of system.debug entries from a previous developer, when I integrated the new and old projects together.
